### ISSUE TITLE
discrepancy between infolist TextEntry and table TextColumn with repeater JSON

### ISSUE THREAD

**Issue by ceejayoz** [laravel-filament/filament#13893]
### Package

filament/filament

### Package Version

v3.2.98

### Laravel Version

v11.20.0

### Livewire Version

v3.5.4

### PHP Version

PHP 8.3.9

### Problem description

When working with a `Repeater`'s JSON array, behavior between `Infolists\Components\TextEntry::make('authors.*.name')` and `Tables\Columns\TextColumn::make('authors.*.name')` varies unexpectedly.

![Screenshot 2024-08-11 at 3 19 48 PM](https://github.com/user-attachments/assets/bb01402c-1c69-4ca2-866e-c8b68b273449)

![Screenshot 2024-08-11 at 3 19 44 PM](https://github.com/user-attachments/assets/8f428716-bd07-4e94-aadd-0929d27d2075)

### Expected behavior

I expected either `authors.name` or `authors.*.name` to work in both spots.

### Steps to reproduce

Full working demonstration in reproduction repository.

Assuming a model with 

```
protected $fillable = [
    'authors',
];

protected $casts = [
    'authors' => 'array',
];
```

and migration with `$table->json('authors')->nullable();`...

Form:

```
Forms\Components\Repeater::make('authors')
    ->schema([
        Forms\Components\TextInput::make('name')
            ->required(),
        Forms\Components\TextInput::make('email')
            ->required()
            ->email(),
    ])
```

Infolist:

```
Infolists\Components\TextEntry::make('authors.name')
    ->label('Authors (authors.name)'),

Infolists\Components\TextEntry::make('authors.*.name')
    ->label('Authors (authors.*.name)'),

Infolists\Components\TextEntry::make('authors.0.name')
    ->label('Authors (authors.0.name)'),
```

Table:

```
Tables\Columns\TextColumn::make('authors.name')
    ->label('Authors (authors.name)'),

Tables\Columns\TextColumn::make('authors.*.name')
    ->label('Authors (authors.*.name)'),

Tables\Columns\TextColumn::make('authors.0.name')
    ->label('Authors (authors.0.name)'),
```

Observe difference between table and infolist display.

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/ceejayoz/filament-repeater-repro

**Comment by danharrin** [comment]
I don't think we really implement `*` functionality anywhere, not sure where you have read that in the docs.

The official way does not require you to do `*`, just `authors.name` on both tables and infolists should produce the same

**Comment by danharrin** [comment]
Sorry, just seen this is a JSON column instead of a relationship (so my prev suggestion of authors.name is wrong). Again though, we don't have support for `*` as far as I know?

**Comment by danharrin** [comment]
In the table, I would suggest doing just `authors` as the column name, so that search will search the JSON array properly, and then doing `->getStateUsing(fn ($record) => collect($record->authors)->pluck('name'))`

**Comment by ceejayoz** [comment]
@danharrin Yeah, my current workaround is `->getStateUsing(fn(Model $record) => $record->authors->pluck('name'))` with a `collection` cast in the model. I know the `*` approach isn't from the docs, but stumbled across it trying to figure out why the usual `authors.name` approach I'd use with a real relationship instead of a JSON column wasn't working.

Happy to close this as resolved, but would y'all consider a feature request to permit `Repeater` JSON to be accessed via an `authors.name` or `authors.*.name` syntax in both `TextEntry` and `TextColumn` components?

**Comment by danharrin** [comment]
Yeah I think we can implement the display part fairly easily (haven't checked, but I assume), but it would probably take quite alot of effort to make it work with sorting/searching. Is that a problem?

**Comment by ceejayoz** [comment]
@danharrin I think display-only would be a wonderful improvement. Searching I think can be left to Laravel Scout in this area.

**Comment by danharrin** [comment]
Hi @ceejayoz, I have implemented that feature in #13963, for the next v3 release

**Comment by ceejayoz** [comment]
@danharrin Thanks!!!
